api_v3/lib/KalturaEntryService.php

```diff
@@ -1551,6 +1551,28 @@ protected function validateEntryScheduleDates(KalturaBaseEntry $entry, entry $db
 			throw new KalturaAPIException(KalturaErrors::INVALID_ENTRY_SCHEDULE_DATES);
 		}
 	}
+	protected function validateNoParentageCycle(KalturaBaseEntry $entry, entry $dbEntry = null)
+	{
+		if (is_null($dbEntry) || is_null($entry->parentEntryId))
+		{
+			return;
+		}
+		$slowParentEntryPointer = $dbEntry;
+		$slowParentEntryPointer->setParentEntryId($entry->parentEntryId);
+		$fastParentEntryPointer = entryPeer::retrieveByPK($slowParentEntryPointer->getParentEntryId());
+		
+		while ($slowParentEntryPointer !== $fastParentEntryPointer)
+		{
+			if ($fastParentEntryPointer == null || !$fastParentEntryPointer->getParentEntryId())
+			{
+				return;
+			}
+			
+			$slowParentEntryPointer = entryPeer::retrieveByPK($slowParentEntryPointer->getParentEntryId());
+			$fastParentEntryPointer = entryPeer::retrieveByPK(entryPeer::retrieveByPK($fastParentEntryPointer->getId())->getId());
```

Comment:
if nothing returned on the first retrieve you will try to get id on a non object